### PR TITLE
Various fixes for VC and BN REST server.

### PR DESCRIPTION
**Pull request by cheatfate**
1. Fix issue when VC i unable to detect error codes properly.
2. Force VC to use `RestPlainResponse` which will allow to print BN response error messages in VC logs.
3. Fix issue when prepareBeaconCommitteeSubnet() unable to perform action when BN is optimistically synced.

### CODE REVIEW

File: beacon_chain/validator_client/api.nim [hunk 1]
```diff
@@ -1366,46 +1451,43 @@ proc submitPoolAttestations*(
...
+                endpoint = node, reason = response.getIndexedErrorMessage()
           node.status = RestBeaconNodeStatus.Offline
           failures.add(ApiNodeFailure.init(node, ApiFailure.Unexpected))
-          false
+          ApiResponse[bool].err(ResponseUnexpectedError)
 
     if res.isErr():
       raise (ref ValidatorApiError)(msg: res.error, data: failures)
-    return true
+    return false
```
Comment (arnetheduck) [inline beacon_chain/validator_client/api.nim]: why false here?

Comment (cheatfate) [reply beacon_chain/validator_client/api.nim]: This is impossible case, but because i can't prove it there is no reason to return successful result.

Comment (arnetheduck) [reply beacon_chain/validator_client/api.nim]: why not `raiseAssert` then?

Comment (arnetheduck) [reply beacon_chain/validator_client/api.nim]: or `doAssert res.isErr`?

**Review by arnetheduck — APPROVED** [review]

File: beacon_chain/validator_client/common.nim [hunk 2]
```diff
@@ -344,6 +348,84 @@ chronicles.expandIt(SyncCommitteeDuty):
...
+      warn "Beacon node has incompatible configuration"
+      node.status = status
+  of RestBeaconNodeStatus.Compatible:
+    if node.status != status:
+      notice "Beacon node is compatible"
+      node.status = status
+  of RestBeaconNodeStatus.NotSynced:
+    if node.status notin {RestBeaconNodeStatus.NotSynced,
+                          RestBeaconNodeStatus.OptSynced}:
+      # When BN returns 503 it doesn't clarify if its opt-synced or not-synced,
```
Comment (arnetheduck) [inline beacon_chain/validator_client/common.nim]: 503 means not-synced - opt-synced is a subset of not-synced: we can either be not-synced (which means both EL and CL are out of sync) or opt-synced (only EL is out of sync). The EL can never be in sync if the CL is out of sync.

**Pull request merged**